src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java

```diff
@@ -55,4 +57,14 @@ public void testIfRetryIsInvokedBeforeListenerButHasToConsiderFailures() {
         assertThat(tla.getFailedTests()).hasSize(1);
         assertThat(tla.getSkippedTests()).hasSize(1);
     }
+
+    @Test(description = "GITHUB-1706")
+    public void testIfRetryIsInvokedWhenTestMethodHasNativeInjection() {
```

Comment:
Could you add a test that mix ORIGIN_DATA_PROVIDER, NATIVE and ORIGIN_XML?

Reply:
@juherr - I have added the required samples, amended the commit and pushed the changes. Please take a look.